### PR TITLE
feat(shell-center-row): add slot for action-bar

### PR DESCRIPTION
**Pull request by asangma**
**Related Issue:** #1007 

## Summary
Add slot for ActionBar.

### CODE REVIEW

**Comment by asangma** [comment]
@kat10140 Let me know if you want to sync re: doc for this.

**Comment by asangma** [comment]
@driskull Just realized I need to add some e2es and stuff.

**Comment by asangma** [comment]
Hokay! Added E2E for action-bar slot and tweaked Readme.

File: src/demos/shell/demo-app-advanced-center-row.html [hunk 1]
```diff
@@ -343,7 +343,7 @@ <h2>Boomer v Millennial - California</h2>
                 <tr>
                   <td class="sml"><calcite-icon icon="square" scale="s"></calcite-icon></td>
                   <td>01</td>
-                  <td>Can't keep Johnny down.</td>
+                  <td>LAST</td>
```
Comment (jcfranco) [inline src/demos/shell/demo-app-advanced-center-row.html:346]: Nitpick: if this is meant to call out being last, can you update this to "This should be last" or something similar?

File: src/components/calcite-shell-center-row/calcite-shell-center-row.scss [hunk 2]
```diff
@@ -60,3 +64,21 @@
...
+.calcite--rtl {
+  ::slotted(calcite-action-bar[position="start"]) {
+    border-right:none;
+    border-left: 1px solid var(--calcite-app-border);
+  }
+  ::slotted(calcite-action-bar[position="end"]) {
+    border-left: none;
+    border-right: 1px solid var(--calcite-app-border);
+  }
+}
```
Comment (jcfranco) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.scss:84]: Could you run Prettier on this file? It should add a trailing newline and ensure space between props and values.

File: src/components/calcite-shell-center-row/calcite-shell-center-row.scss [hunk 3]
```diff
@@ -60,3 +64,21 @@
   width: 100%;
   height: 100%;
 }
+
+::slotted(calcite-action-bar[position="start"]) {
+  border-right: 1px solid var(--calcite-app-border);
+}
+::slotted(calcite-action-bar[position="end"]) {
+  border-left: 1px solid var(--calcite-app-border);
```
Comment (jcfranco) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.scss:72]: Can you store `1px solid var(--calcite-app-border)` in a var and reuse?

File: src/components/calcite-shell-center-row/calcite-shell-center-row.tsx [hunk 4]
```diff
@@ -33,19 +35,50 @@ export class CalciteShellCenterRow {
...
+    const { el } = this;
+    const rtl = getElementDir(el) === "rtl";
+    
+    const contentNode = (
+      <div class={{ [CSS.content]: true, [CSS_UTILITY.rtl]: rtl }}>
+        <slot />
+      </div>
     );
+
+    const actionBar = getSlotted<HTMLCalciteActionBarElement>(el, SLOTS.actionBar);
```
Comment (driskull) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.tsx:62]: suggestion:

```tsx
const actionBarNode = getSlotted<HTMLCalciteActionBarElement>(el, SLOTS.actionBar) ? (
      <div class={{ [CSS.actionBarContainer]: true, [CSS_UTILITY.rtl]: rtl }}>
        <slot name={SLOTS.actionBar} />
      </div>
    ) : null;

File: src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts [hunk 5]
```diff
@@ -21,6 +24,40 @@ describe("calcite-shell-center-row", () => {
...
+      </calcite-action-bar>
+    </calcite-shell-center-row>
+    `
+    await page.setContent(pageContent);
+
+    const actionBarContainer = await page.find(`calcite-shell-center-row >>> .${CSS.actionBarContainer}`);
+
+    expect(actionBarContainer).not.toBeNull();
+  });
+
```
Comment (jcfranco) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts:53]: This needs a test for the start/end positioning of action-bar.

File: src/components/calcite-shell-center-row/calcite-shell-center-row.tsx [hunk 6]
```diff
@@ -33,19 +35,50 @@ export class CalciteShellCenterRow {
...
+      return <Host>{contentNode}</Host>;
+    }
+
+    const actionBarNode = (
+      <div class={{ [CSS.actionBarContainer]: true, [CSS_UTILITY.rtl]: rtl }}>
+        <slot name={SLOTS.actionBar} />
+      </div>
+    );
+
+    const children = [contentNode];
```
Comment (driskull) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.tsx:74]: suggestion: 

```tsx
const children = [actionBarNode, contentNode];

if(actionBar.position === "end"){
children.reverse();
}

```

File: src/components/calcite-shell-center-row/calcite-shell-center-row.tsx [hunk 7]
```diff
@@ -33,19 +35,50 @@ export class CalciteShellCenterRow {
...
+    
+    const contentNode = (
+      <div class={{ [CSS.content]: true, [CSS_UTILITY.rtl]: rtl }}>
+        <slot />
+      </div>
     );
+
+    const actionBar = getSlotted<HTMLCalciteActionBarElement>(el, SLOTS.actionBar);
+
+    if (!actionBar) {
```
Comment (driskull) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.tsx:64]: suggestion: remove this block.

File: src/demos/shell/demo-app-advanced-center-row.html (hunk 1 above)
Comment (asangma) [reply src/demos/shell/demo-app-advanced-center-row.html:346]: 👍

**Comment by asangma** [comment]
Thanks @jcfranco. I'll do all the things.

File: src/components/calcite-shell-center-row/calcite-shell-center-row.tsx (hunk 6 above)
Comment (asangma) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.tsx:74]: Wouldn't I need to nest this in a conditional for whether or not there's an action-bar?

Comment (asangma) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.tsx:74]: Oh...I see. Lemme try dat.

Comment (driskull) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.tsx:74]: I don't think so. it won't render anything null

**Comment by asangma** [comment]
@jcfranco @driskull Please give it another review. 🙏

File: src/components/calcite-action-bar/calcite-action-bar.tsx [hunk 8]
```diff
@@ -73,7 +73,7 @@ export class CalciteActionBar {
   /**
    * Arranges the component depending on the elements 'dir' property.
    */
-  @Prop({ reflect: true }) position: CalcitePosition;
+  @Prop({ reflect: true }) position: CalcitePosition = "start";
```
Comment (driskull) [inline src/components/calcite-action-bar/calcite-action-bar.tsx:76]: Do we need this to have a value or can we assume `start` if not present?

File: src/components/calcite-shell-center-row/calcite-shell-center-row.scss [hunk 9]
```diff
@@ -10,6 +10,8 @@
   --calcite-app-shell-center-row-height-small: 33%;
   --calcite-app-shell-center-row-height-medium: 70%;
   --calcite-app-shell-center-row-height-large: 100%;
+
+  --calcite-app-shell-center-row-border: 1px solid var(--calcite-app-border);
```
Comment (jcfranco) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.scss:14]: Sorry, I meant to suggest a sass var for this.

File: src/components/calcite-action-bar/calcite-action-bar.tsx (hunk 8 above)
Comment (asangma) [reply src/components/calcite-action-bar/calcite-action-bar.tsx:76]: I am using that attribute in the styles in center-row.

File: src/components/calcite-shell-center-row/calcite-shell-center-row.scss (hunk 9 above)
Comment (asangma) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.scss:14]: Oh. Is it better as a sass var?

File: src/components/calcite-action-bar/calcite-action-bar.tsx (hunk 8 above)
Comment (asangma) [reply src/components/calcite-action-bar/calcite-action-bar.tsx:76]: Although...yeah. I see. I'll update.

Comment (asangma) [reply src/components/calcite-action-bar/calcite-action-bar.tsx]: @driskull Done. Thanks.

File: src/components/calcite-shell-center-row/calcite-shell-center-row.scss (hunk 9 above)
Comment (asangma) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.scss:14]: @jcfranco I'm down to do that but we would wanna update all the sass files to use that var, yeah?

**Pull request merged**